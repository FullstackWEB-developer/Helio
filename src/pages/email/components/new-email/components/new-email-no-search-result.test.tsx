import i18n from '../../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import NewEmailNoSearchResult from './new-email-no-search-result';
import Router from "react-router-dom";
import EmailProvider from '@pages/email/context/email-context';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("New Email No Search Result tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders new-email-no-search-result correctly", async () => {
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })

        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <EmailProvider>
                <NewEmailNoSearchResult searchTerm={""}/>
            </EmailProvider>            
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})