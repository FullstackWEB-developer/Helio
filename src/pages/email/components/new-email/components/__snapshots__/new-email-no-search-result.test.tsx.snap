// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Email No Search Result tests renders new-email-no-search-result correctly 1`] = `
<DocumentFragment>
  <div
    class="pt-8 pl-6 body2"
  >
    search.search_results.empty
  </div>
</DocumentFragment>
`;
